src/providers/conversion/ConversionRateProvider.ts

```diff
@@ -0,0 +1,21 @@
+// We are keeping an organized enum of all currencies to allow for easier refactoring should
+// we decide to change the way that metaplex names currencies internally
+export enum Currency {
+  USD = 'usd',
+  EUR = 'eur',
+  AR = 'ar',
+  SOL = 'sol',
+}
+
+export type ConversionRatePair = {
+  base: Currency;
+  quote: Currency;
+  rate: number;
+};
+
+export abstract class ConversionRateProvider {
+  static getRate: (
+    from: Currency | Currency[],
+    to: Currency | Currency[],
+  ) => Promise<ConversionRatePair[]>;
+}
```

Comment:
This is done to remove the hard requirement of Coingecko for minting but still keep the ability to go from A-Z of minting process within the scope of Metaplex JS. 

The idea is that we provide both an abastract class for a conversion rate provider (the minting process does lots of lamport calculation there), but only provide implementation for Coingecko. 

This can be easily thrown out & replaced by the developer by any provider he trusts.